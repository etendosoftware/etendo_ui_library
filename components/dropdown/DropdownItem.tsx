import React from 'react';
import {Text, TouchableOpacity} from 'react-native';

import {DropdownStyleVariant} from './Dropdown.styles';
import {DropdownItemProps} from './Dropdown.types';
import {getSizeText, getStyle} from './DropdownItem.styles';

const DropdownItem = ({
  item,
  onPress,
  index,
  toggleItem,
  sizeText,
}: DropdownItemProps) => {
  return (
    <TouchableOpacity
      style={getStyle(toggleItem, index)}
      onPress={() => {
        onPress();
      }}
      key={item.key}
      activeOpacity={1}
    >
      <Text
        key={item.key}
        style={[
          {fontSize: getSizeText(sizeText)},
          DropdownStyleVariant.primary.textOption,
        ]}
      >
        {item.name}
      </Text>
    </TouchableOpacity>
  );
};

export default DropdownItem;
