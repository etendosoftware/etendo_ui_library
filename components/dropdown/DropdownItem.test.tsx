import React from 'react';
import renderer from 'react-test-renderer';
import DropdownItem from './DropdownItem';

const item = {name: 'User', route: '', key: 'user'};

describe('Running Test for DropdownItem', () => {
  it('Check DropdownItem Disabled', () => {
    const tree = renderer
      .create(
        <DropdownItem
          item={item}
          onPress={() => {}}
          index={0}
          toggleItem={0}
          sizeText={'medium'}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
