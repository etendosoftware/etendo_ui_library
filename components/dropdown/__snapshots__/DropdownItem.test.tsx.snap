// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running Test for DropdownItem Check DropdownItem Disabled 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FAD614",
      "borderBottomColor": "#D3D6E1",
      "borderBottomWidth": 1.5,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
        },
        Object {
          "fontWeight": "500",
          "margin": 10,
          "textAlign": "left",
        },
      ]
    }
  >
    User
  </Text>
</View>
`;
